lib/wanda_web/controllers/executions_controller.ex

```diff
@@ -0,0 +1,34 @@
+defmodule WandaWeb.ExecutionsController do
+  use WandaWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Wanda.Results
+  alias WandaWeb.Schemas.CheckExecutionResponse
+
+  plug OpenApiSpex.Plug.CastAndValidate, json_render_error_v2: true
+
+  operation :list_checks_executions,
+    summary: "List check executions",
+    parameters: [
+      group_id: [
+        in: :query,
+        description: "Group ID to filter upon",
+        type: :string,
+        example: "00000000-0000-0000-0000-000000000001"
+      ],
+      page: [in: :query, description: "Page", type: :integer, example: 3],
+      items_per_page: [in: :query, description: "Items per page", type: :integer, example: 20]
+    ],
+    responses: [
+      ok: {"Check executions response", "application/json", CheckExecutionResponse}
+    ]
+
+  def list_checks_executions(conn, params) do
+    page = Map.get(params, :page, 1)
+    items_per_page = Map.get(params, :items_per_page, 10)
+    group_id = Map.get(params, :group_id)
+
+    results = Results.get_execution_results(group_id, page, items_per_page)
```

Comment:
we have views for that, i think we can iterate adding the views in a new PR. 

https://hexdocs.pm/phoenix/views.html

Reply:
@arbulu89 I'm super glad you noticed that, yeah it's something we have to do but I'd like to do that in a follow-up PR because I was afraid this PR was getting really huge

Reply:
@dottorblaster Fine for me them! We can follow up.